packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverPageCount/TeachingPopoverPageCount.types.ts

```diff
@@ -1,17 +1,49 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 
 export type TeachingPopoverPageCountSlots = {
-  root: Slot<'div'>;
+  /**
+   * The element wrapping the carousel pagination. By default this is a div,
+   * it may contain icons or text depending on TeachingPopoverPageCountStyle
+   */
+  root: NonNullable<Slot<'div'>>;
+
+  /**
+   * The element for each carousel page in 'icon' mode.
+   */
+  carouselIcon: NonNullable<Slot<'button'>>;
+
+  /**
+   * The element for each carousel page in 'icon' mode.
+   */
+  carouselSelectedIcon: NonNullable<Slot<'button'>>;
+};
+
+export type TeachingPopoverPageCountStyle = 'text' | 'icon';
+
+export type TeachingPopoverPageCountState = ComponentState<TeachingPopoverPageCountSlots> & {
+  /**
+   * The current carousel page.
+   */
+  currentPage: number;
+  /**
+   * The total number of carousel pages, controlled by children within carousel itself.
+   */
+  totalPages: number;
+  /**
+   * Allows the carousel pagination to update the current page based on icon interaction.
+   */
+  setCurrentPage: (index: number) => void;
+  /**
+   * The style in which the page count is rendered.
+   */
+  countStyle: TeachingPopoverPageCountStyle;
 };
 
-/**
- * TeachingPopoverPageCount Props
- */
-export type TeachingPopoverPageCountProps = ComponentProps<TeachingPopoverPageCountSlots> & {};
+// For localization or customization, users may want to modify this for their own purposes
+export type TeachingPopoverPageCountChildRenderFunction = (currentPage: number, totalPages: number) => React.ReactNode;
 
-/**
- * State used in rendering TeachingPopoverPageCount
- */
-export type TeachingPopoverPageCountState = ComponentState<TeachingPopoverPageCountSlots>;
-// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from TeachingPopoverPageCountProps.
-// & Required<Pick<TeachingPopoverPageCountProps, 'propName'>>
+export type TeachingPopoverPageCountProps = ComponentProps<Partial<TeachingPopoverPageCountSlots>> &
```

Comment:
```suggestion
export type TeachingPopoverPageCountProps = ComponentProps<TeachingPopoverPageCountSlots> &
```

Reply:
I'll need to modify this as a part of moving the carousel icons into their own slots, for some reason making the icons optional props causes type issues with how we are generating an array of them.

Reply:
Resolution tracked by #30127.